olp-cpp-sdk-core/src/cache/DiskCache.cpp

```diff
@@ -216,13 +233,17 @@ OpenResult DiskCache::Open(const std::string& data_path,
     status = leveldb::DB::Open(open_options, versioned_data_path, &db);
   }
 
-  // If the database is r/w and corrupted, attempt to repair & reopen
-  if ((status.IsCorruption() || status.IsIOError()) &&
-      RepairCache(versioned_data_path) == true) {
-    status = leveldb::DB::Open(open_options, versioned_data_path, &db);
-    if (status.ok()) {
-      database_.reset(db);
-      return OpenResult::Repaired;
+  bool repaired = false;
+  if (status.IsCorruption() || status.IsIOError()) {
+    if (is_read_only) {
+      OLP_SDK_LOG_ERROR_F(kLogTag, "Open: cache corrupted, error='%s'",
```

Comment:
Maybe also add path of the cache?

Reply:
The cache path will be reported by DefaultCacheImp, should we duplicate it here?

Reply:
I would not trust anybody. Every component logs for itself. It is not a issue to have duplicate data for such important logs.